assemble.py

```diff
@@ -336,6 +390,15 @@ def get_args():
             target.lmp_version = release_info.lmp_version
             if args.app_type == 'restorable' or (not args.app_type and release_info.lmp_version > 84):
                 logger.info('Fetching Restorable Apps...')
+                previously_fetched_apps_uri, previously_fetched_apps \
+                    = check_and_get_fetched_apps_uri(target, args.app_shortlist)
+                if previously_fetched_apps_uri:
+                    target_apps_dir = os.path.join(apps_root_dir, target.name)
+                    logger.info("Fetching the app archive from the publish run; uri:"
+                                f" {previously_fetched_apps_uri}, apps: {previously_fetched_apps}")
+                    get_and_extract_fetched_apps(previously_fetched_apps_uri, args.token,
+                                                 target_apps_dir)
+                    logger.info(f"The fetched app archive is extracted to {target_apps_dir}")
```

Comment:
double checking that we don't need an `else` fetch_restorable_apps

Reply:
No `else` is needed. A set of apps fetched for offline update in the publish run can differ from a set of apps for preloading.
So, the current implementation reuses the publish run apps only if it is a sub-set of the apps for preloading. So, we still need to call the fetch to pull the diff. If the sets are the same then we could avoid the fetch here, but I decided it's not worth of additional `if previously_fetched_apps == apps_to_fetch then skip fetching` because `skopeo` is smart enough not to pull blobs if they already present locally.

When/If I move `ci-scripts` from `skopeo` to `composectl` then I'll improve this logic, so the publish run apps are re-used even if its set is not subset of apps for preloading. With `skopeo` it's a bit cumbersome to implement (think about shared blobs folder).